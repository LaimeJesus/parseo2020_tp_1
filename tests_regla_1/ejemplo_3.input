fun f
fun g : x -> y
fun h
    A -> B
fun i : x -> y
    A -> B