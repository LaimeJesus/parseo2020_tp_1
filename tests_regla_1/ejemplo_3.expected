[
   "program",
   [
      [
         "fun",
         "f",
         [
            "sig",
            [],
            "_"
         ],
         [
            "pre",
            [
               "true"
            ]
         ],
         [
            "post",
            [
               "true"
            ]
         ],
         []
      ],
      [
         "fun",
         "g",
         [
            "sig",
            [
               "x"
            ],
            "y"
         ],
         [
            "pre",
            [
               "true"
            ]
         ],
         [
            "post",
            [
               "true"
            ]
         ],
         []
      ],
      [
         "fun",
         "h",
         [
            "sig",
            [
               "_"
            ],
            "_"
         ],
         [
            "pre",
            [
               "true"
            ]
         ],
         [
            "post",
            [
               "true"
            ]
         ],
         [
            [
               "rule",
               [
                  [
                     "pcons",
                     "A",
                     []
                  ]
               ],
               [
                  "cons",
                  "B",
                  []
               ]
            ]
         ]
      ],
      [
         "fun",
         "i",
         [
            "sig",
            [
               "x"
            ],
            "y"
         ],
         [
            "pre",
            [
               "true"
            ]
         ],
         [
            "post",
            [
               "true"
            ]
         ],
         [
            [
               "rule",
               [
                  [
                     "pcons",
                     "A",
                     []
                  ]
               ],
               [
                  "cons",
                  "B",
                  []
               ]
            ]
         ]
      ]
   ],
   []
]
